### PR TITLE
[Cache] Test & tweak CacheItem::validateKey()

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | 3.1 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Ping @javiereguiluz @stof

### CODE REVIEW

File: src/Symfony/Component/Cache/Tests/CacheItemTest.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Tests;
+
+use Symfony\Component\Cache\CacheItem;
+
+class CacheItemTest extends \PHPUnit_Framework_TestCase
+{
+    public function testValideKey()
```
Comment (stof) [inline src/Symfony/Component/Cache/Tests/CacheItemTest.php]: ``ValidKey``

File: src/Symfony/Component/Cache/Tests/CacheItemTest.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        $this->assertNull(CacheItem::validateKey('foo'));
+    }
+
+    /**
+     * @dataProvider provideInvalideKey
+     * @expectedException Symfony\Component\Cache\Exception\InvalidArgumentException
+     * @expectedExceptionMessage Cache key
+     */
+    public function testInvalideKey($key)
```
Comment (stof) [inline src/Symfony/Component/Cache/Tests/CacheItemTest.php]: ``InvalidKey``

File: src/Symfony/Component/Cache/Tests/CacheItemTest.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            array('/'),
+            array('\\'),
+            array('@'),
+            array(':'),
+            array(true),
+            array(null),
+            array(1),
+            array(1.1),
+            array(array()),
+            array(fopen(__FILE__, 'rb')),
```
Comment (stof) [inline src/Symfony/Component/Cache/Tests/CacheItemTest.php]: wouldn't this leak the resource ?

Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Tests/CacheItemTest.php]: AFAIK, resources are autoclosed when not used anymore

File: src/Symfony/Component/Cache/Tests/CacheItemTest.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Tests/CacheItemTest.php]: fixed

**Comment by javiereguiluz** [comment]
👍  It looks very nice to me.

@nicolas-grekas thanks for this PR ... and thanks for your patience 🙏

**Comment by nicolas-grekas** [comment]
Thanks to you @javiereguiluz , you had a good point, sorry for merging too fast today :)

**Pull request merged**